### PR TITLE
Added z_shieldcoinbase contextual menu option for z-addresses.

### PR DESCRIPTION
**Pull request by donshin**
The contextual menu for z-addresses listed in the "My Addresses" tab now features a menu option, "Shield all coinbase to this z-address"; upon option selection, the wallet UI will display a text response that confirms how much coinbase funds from however many UTXOs were successfully shielded to that z-address.

### CODE REVIEW

File: src/main/java/org/btcprivate/wallets/fullnode/util/Util.java [hunk 1]
```diff
@@ -320,7 +320,8 @@ public static String convertGroupPhraseToZPrivateKey(String phrase)
     public static boolean isZAddress(String address)
     {
         return (address != null) &&
-                address.startsWith("zk") &&
+                (address.startsWith("zk")) &&
+//                (address.startsWith("zk") || address.startsWith("zz") || address.startsWith("zt")) &&
```
Comment (Pega88) [inline src/main/java/org/btcprivate/wallets/fullnode/util/Util.java:324]: did this sneak in? or do we need this for testnets to comment it out?

Comment (donshin) [reply src/main/java/org/btcprivate/wallets/fullnode/util/Util.java:324]: The commented out line was kept in simply for convenience- for future testing w/ regtest/testnet addresses. I'm fine with leaving it in or taking it out as well.

File: src/main/java/org/btcprivate/wallets/fullnode/ui/AddressTable.java [hunk 2]
```diff
@@ -96,6 +117,87 @@ public void actionPerformed(ActionEvent e)
                 }
             }
         });
-    } // End constructor
+		return menuItem;
+    }
+    
+    protected JMenuItem instantiateShieldAllCoinbaseMenuItem() {
+        JMenuItem shieldAllCoinbaseFundsMenuItem = new JMenuItem("Shield all coinbase to this z-address");
```
Comment (ch4ot1c) [inline src/main/java/org/btcprivate/wallets/fullnode/ui/AddressTable.java]: coinbases

File: src/main/java/org/btcprivate/wallets/fullnode/ui/AddressTable.java [hunk 3]
```diff
@@ -96,6 +117,87 @@ public void actionPerformed(ActionEvent e)
...
+                        // Lock the wallet again
+                        if (bEncryptedWallet)
+                        {
+                            caller.lockWallet();
+                        }
+
+                        JOptionPane.showMessageDialog(
+                                AddressTable.this.getRootPane().getParent(),
+                                "Coinbase funds in the amount of " + shieldCoinbaseResponse.shieldedValue + " BTCP from " + shieldCoinbaseResponse.shieldedUTXOs + " UTXO" + (shieldCoinbaseResponse.shieldedUTXOs == 1 ? "" : "s") + " were shielded to the following z-address:\n" + address + "\nPlease Refresh to update balances.\n",
+                                "Shield All Coinbase", JOptionPane.INFORMATION_MESSAGE);
```
Comment (ch4ot1c) [inline src/main/java/org/btcprivate/wallets/fullnode/ui/AddressTable.java]: Coinbases

File: src/main/java/org/btcprivate/wallets/fullnode/ui/AddressTable.java [hunk 4]
```diff
@@ -96,6 +117,87 @@ public void actionPerformed(ActionEvent e)
...
+						// TODO: trigger refresh of window
+
+                    } catch (Exception ex)
+                    {
+                        Log.error("Unexpected error: ", ex);
+                        JOptionPane.showMessageDialog(
+                                AddressTable.this.getRootPane().getParent(),
+                                "Error:" + "\n" +
+                                        ex.getMessage() + "\n\n",
+                                "Error in shielding all coinbase!",
```
Comment (ch4ot1c) [inline src/main/java/org/btcprivate/wallets/fullnode/ui/AddressTable.java]: coinbases

**Pull request merged**